googleapis/sdk-platform-java
# feat: dynamically determine protobuf version in build.gradle

Fixes #1757.

### Steps to verify this PR:
1. find a self-service client, e.g., `google-maps-fleetengine`.
2. checkout this branch, assume the path of repo root is `/path/sdkplatform-java`.
3. in `googleapis` repository, run `bazelisk build` to build the client library with `gapic-generator-java` and `gax-java` replaced with local repository.
  For example, `bazelisk build //google/maps/fleetengine/v1:google-maps-fleetengine-v1-java --override_repository=gapic_generator_java=/path/sdk-platform-java/ --override_repository=com_google_api_gax_java=/path/sdk-platform-java/gax-java/`
4. in `googleapis` repository, unzip the generated tar file to `googleapis` root directory.
   For example, `tar -xf bazel-bin/google/maps/fleetengine/v1/google-maps-fleetengine-v1-java.tar.gz`
5. find protobuf dependency, `com.google.protobuf:protobuf-java` in `proto-*/build.gradle` and make sure the version is the same as protobuf version defined in `WORKSPACE` in `gooleapis`.
6. as an alternative, go to the client library directory (e.g., `google-maps-fleetengine-v1-java/`), and run `./gradlew build`.
